api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -47,18 +52,69 @@
   private static final String SAM_STATUS_NAME = "Sam";
   private static final String RAWLS_STATUS_NAME = "Rawls";
   private static final String GOOGLE_BUCKETS_STATUS_NAME = "GoogleBuckets";
+  private static final int MAX_ATTEMPTS = 3;
 
   @Autowired
   public FireCloudServiceImpl(Provider<WorkbenchConfig> configProvider,
       Provider<ProfileApi> profileApiProvider,
       Provider<BillingApi> billingApiProvider,
       Provider<GroupsApi> groupsApiProvider,
-      Provider<WorkspacesApi> workspacesApiProvider) {
+      Provider<WorkspacesApi> workspacesApiProvider,
+      Sleeper sleeper) {
     this.configProvider = configProvider;
     this.profileApiProvider = profileApiProvider;
     this.billingApiProvider = billingApiProvider;
     this.groupsApiProvider = groupsApiProvider;
     this.workspacesApiProvider = workspacesApiProvider;
+    this.sleeper = sleeper;
+  }
+
+  private static boolean isServiceUnavailableException(Exception e) {
+    if (e instanceof ApiException) {
+      int code = ((ApiException) e).getCode();
+      return (code > 500 && code <= 600);
```

Comment:
nit: < 600 ? We probably should give up indefinitely if Firecloud starts returning 600s

Reply:
Done.